File: java-sdk/radar-schemas-tools/src/main/java/org/radarcns/schema/specification/connector/ConnectorSource.java
package org.radarcns.schema.specification.connector;

import com.fasterxml.jackson.annotation.JsonProperty;
import org.radarcns.schema.Scope;
import org.radarcns.schema.specification.DataProducer;
import org.radarcns.schema.specification.DataTopic;

import java.util.List;

/**
 * Data producer for third-party connectors. This data topic does not register schemas to the schema
 * registry by default, since Kafka Connect will do that itself. To enable auto-registration, set
 * the {@code register_schema} property to {@code true}.
 */
public class ConnectorSource extends DataProducer<DataTopic> {
    @JsonProperty
    private List<DataTopic> data;

    @JsonProperty
    private String vendor;

    @JsonProperty
    private String model;

    @JsonProperty
    private String version;

    public ConnectorSource() {
        registerSchema = false;
    }

    @Override
    public List<DataTopic> getData() {
        return data;
    }

    @Override
    public Scope getScope() {
        return Scope.CONNECTOR;
    }

    public String getVendor() {
        return vendor;
    }

    public String getModel() {
        return model;
    }

    public String getVersion() {
        return version;
    }
}

File: java-sdk/radar-schemas-core/src/main/java/org/radarbase/schema/specification/connector/ConnectorSource.kt
package org.radarbase.schema.specification.connector

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL
import com.fasterxml.jackson.annotation.JsonProperty
import org.radarbase.config.OpenConfig
import org.radarbase.schema.Scope
import org.radarbase.schema.Scope.CONNECTOR
import org.radarbase.schema.specification.DataProducer
import org.radarbase.schema.specification.DataTopic

/**
 * Data producer for third-party connectors. This data topic does not register schemas to the schema
 * registry by default, since Kafka Connect will do that itself. To enable auto-registration, set
 * the `register_schema` property to `true`.
 */
@JsonInclude(NON_NULL)
@OpenConfig
class ConnectorSource : DataProducer<DataTopic>() {
    @JsonProperty
    override var data: MutableList<DataTopic> = mutableListOf()

    @JsonProperty
    var vendor: String? = null

    @JsonProperty
    var model: String? = null

    @JsonProperty
    var version: String? = null

    override var registerSchema = false

    override val scope: Scope = CONNECTOR
}
